Data_Structures_Answers.md

```diff
@@ -1,18 +1,26 @@
 Add your answers to the questions below.
 
 1. What is the runtime complexity of your ring buffer's `append` method?
+_O(1)_
 
 2. What is the space complexity of your ring buffer's `append` function?
+_O(1)_
 
 3. What is the runtime complexity of your ring buffer's `get` method?
+_O(n)_
 
 4. What is the space complexity of your ring buffer's `get` method?
+_O(n)_
 
 
 5. What is the runtime complexity of the provided code in `names.py`?
+_O(n^2)_
 
 6. What is the space complexity of the provided code in `names.py`?
+_O(1)_
 
 7. What is the runtime complexity of your optimized code in `names.py`?
+_O(n)_
```

Comment:
consider that you are using a Binary tree for look-up, and each lookup in a BST is O(log(n)) complexity.  Since we do this n times, this will actually be O(nlog(n)) overall.